Adding dev-perl/Feature-Try-Compat

einhverfr opened the pull request:
This module tries to use Perl 5.34's try/catch syntax and if not available falls back to Syntax::Keyword::Try.  This module is becoming more heavily used as people look towards core try/catch syntax while still having to support prior versions of Perl.

Feedback welcome.

einhverfr commented:
I have now corrected the missing sign-off tag.

I would like feedback on this package because then I can submit a PR with everything necessary for the eventual sending of all of LedgerSMB, but don't want to waste people's time so would like to get a few small packages in first, and then send a mega-PR with the rest, carefully structured. by top-level dependency.

thesamesam commented on dev-perl/Feature-Compat-Try/Feature-Compat-Try-0.05.ebuild:
```diff
@@ -0,0 +1,20 @@
+
+inherit perl-module
+
+DESCRIPTION="C<Feature::Compat::Try> - make C<try/catch> syntax available"
```
This might be a little bit long, but it could be OK for now.

thesamesam commented on dev-perl/Feature-Compat-Try/Feature-Compat-Try-0.05.ebuild:
```diff
@@ -0,0 +1,20 @@
+EAPI=8
+
+DIST_AUTHOR="PEVANS"
+DIST_VERSION="0.05"
```
Do we need this? (Does pkgcheck warn without it?)

See https://wiki.gentoo.org/wiki/Project:Perl/How_to_version-bump_dev-perl#pkgcheck btw.

thesamesam reviewed (commented):
Looks good as a start! Make sure you use 'pkgdev commit' to get the commit messages right (or type it manually if you want, format for summary is `cat/pkg: summary`).

einhverfr commented:
Thanks very much for your review.  I will make these changes.

So capturing the feedback in general:
1.  Shorter descriptions would be better.
2.  I need to be careful about RDEPEND/BDEPEND and make sure I split these out properly
3.  I should check if DIST_VERSION is required and remove otherwise
4.  I should use pkgdev --commit.

Working on this now.

einhverfr commented:
I think if this looks good, it would be good enough for me to start the process of moving all of the LedgerSMB 1.9 dependencies and the main app to a mega-PR.  Unfortunately LedgerSMB 1.9's build system is not great so my ebuild in the overlay is... amusing there and may take significant review.

akhuettel commented on dev-perl/PGObject-Type-DateTime/PGObject-Type-DateTime-2.000002.ebuild:
```diff
@@ -0,0 +1,20 @@
+LICENSE="BSD-2"
+SLOT="0"
+KEYWORDS="~amd64"
+IUSE=""
```
remove

akhuettel commented on dev-perl/PGObject-Type-DateTime/PGObject-Type-DateTime-2.000002.ebuild:
```diff
@@ -0,0 +1,20 @@
+
+RDEPEND="dev-perl/DateTime-TimeZone
+	dev-perl/DateTime
+	>=dev-perl/PGObject-2.000001
```
I guess you mean 2.0.1

akhuettel commented on dev-perl/PGObject-Type-DateTime/PGObject-Type-DateTime-2.000002.ebuild:
```diff
@@ -0,0 +1,20 @@
+RDEPEND="dev-perl/DateTime-TimeZone
+	dev-perl/DateTime
+	>=dev-perl/PGObject-2.000001
+	dev-lang/perl"
```
remove

akhuettel commented on dev-perl/PGObject-Util-DBAdmin/PGObject-Util-DBAdmin-1.6.1.ebuild:
```diff
@@ -0,0 +1,23 @@
+LICENSE="BSD-2"
+SLOT="0"
+KEYWORDS="~amd64"
+IUSE=""
```
remove

akhuettel commented on dev-perl/PGObject-Util-DBAdmin/PGObject-Util-DBAdmin-1.6.1.ebuild:
```diff
@@ -0,0 +1,23 @@
+	dev-perl/DBI
+	dev-perl/Capture-Tiny
+	dev-perl/Moo
+	dev-lang/perl"
```
remove

akhuettel commented on dev-perl/PGObject-Util-DBMethod/PGObject-Util-DBMethod-1.01.000.ebuild:
```diff
@@ -0,0 +1,17 @@
+EAPI=8
+
+DIST_AUTHOR="EHUELS"
+
```
DIST_VERSION=1.01.000

and the ebuild is 1.1.0
dev-perl/Gentoo-PerlMod-Version  <-- this gives you the definitive values

akhuettel commented on dev-perl/PGObject-Util-DBMethod/PGObject-Util-DBMethod-1.01.000.ebuild:
```diff
@@ -0,0 +1,17 @@
+LICENSE="BSD-2"
+SLOT="0"
+KEYWORDS="~amd64"
+IUSE=""
```
remove

akhuettel commented on dev-perl/PGObject-Util-DBMethod/PGObject-Util-DBMethod-1.01.000.ebuild:
```diff
@@ -0,0 +1,17 @@
+KEYWORDS="~amd64"
+IUSE=""
+
+DEPEND="dev-lang/perl"
```
remove

akhuettel commented on dev-perl/PGObject/PGObject-2.3.2.ebuild:
```diff
@@ -0,0 +1,21 @@
+LICENSE="BSD-2"
+SLOT="0"
+KEYWORDS="~amd64"
+IUSE=""
```
remove

akhuettel commented on dev-perl/PGObject/PGObject-2.3.2.ebuild:
```diff
@@ -0,0 +1,21 @@
+
+RDEPEND="dev-perl/Try-Tiny
+	dev-perl/DBD-Pg
+	dev-lang/perl"
```
remove

akhuettel commented on dev-perl/Plack-Builder-Conditionals/Plack-Builder-Conditionals-0.05.ebuild:
```diff
@@ -0,0 +1,22 @@
+EAPI=8
+
+DIST_AUTHOR="KAZEBURO"
+
```
DIST_VERSION=0.05

and the ebuild is 0.50.0

akhuettel commented on dev-perl/Plack-Builder-Conditionals/Plack-Builder-Conditionals-0.05.ebuild:
```diff
@@ -0,0 +1,22 @@
+
+DESCRIPTION="Plack::Builder match-based rules"
+
+LICENSE="|| ( Artistic GPL-1 GPL-2 GPL-3 )"
```
remove

akhuettel commented on dev-perl/Plack-Builder-Conditionals/Plack-Builder-Conditionals-0.05.ebuild:
```diff
@@ -0,0 +1,22 @@
+LICENSE="|| ( Artistic GPL-1 GPL-2 GPL-3 )"
+SLOT="0"
+KEYWORDS="~amd64"
+IUSE=""
```
rremove

akhuettel commented on dev-perl/Plack-Builder-Conditionals/Plack-Builder-Conditionals-0.05.ebuild:
```diff
@@ -0,0 +1,22 @@
+RDEPEND=">=dev-perl/Plack-1.4.800
+	dev-perl/List-MoreUtils
+	dev-perl/Net-CIDR-Lite
+	dev-lang/perl"
```
remove

akhuettel commented on dev-perl/Plack-Middleware-ReverseProxy/Plack-Middleware-ReverseProxy-0.16.ebuild:
```diff
@@ -0,0 +1,18 @@
+EAPI=8
+
+DIST_AUTHOR="MIYAGAWA"
+
```
DIST_VERSION=0.16

and the ebuild is 0.160.0

akhuettel commented on dev-perl/Plack-Middleware-ReverseProxy/Plack-Middleware-ReverseProxy-0.16.ebuild:
```diff
@@ -0,0 +1,18 @@
+
+DESCRIPTION="Supports app to run as a reverse proxy backend"
+
+LICENSE="|| ( Artistic GPL-1 GPL-2 GPL-3 )"
```
remove

akhuettel commented on dev-perl/Plack-Middleware-ReverseProxy/Plack-Middleware-ReverseProxy-0.16.ebuild:
```diff
@@ -0,0 +1,18 @@
+LICENSE="|| ( Artistic GPL-1 GPL-2 GPL-3 )"
+SLOT="0"
+KEYWORDS="~amd64"
+IUSE=""
```
remove

akhuettel commented on dev-perl/Plack-Middleware-ReverseProxy/Plack-Middleware-ReverseProxy-0.16.ebuild:
```diff
@@ -0,0 +1,18 @@
+IUSE=""
+
+RDEPEND=">=dev-perl/Plack-1.4.800
+	dev-lang/perl"
```
remove

akhuettel commented on dev-perl/Plack-Request-WithEncoding/Plack-Request-WithEncoding-0.14.ebuild:
```diff
@@ -0,0 +1,23 @@
+EAPI=8
+
+DIST_AUTHOR="MOZNION"
+
```
DIST_VERSION=0.14

and the ebuild is 0.140.0

akhuettel commented on dev-perl/Plack-Request-WithEncoding/Plack-Request-WithEncoding-0.14.ebuild:
```diff
@@ -0,0 +1,23 @@
+
+DESCRIPTION="Subclass of L<Plack::Request> which supports encoded requests."
+
+LICENSE="|| ( Artistic GPL-1 GPL-2 GPL-3 )"
```
remove

akhuettel commented on dev-perl/Plack-Request-WithEncoding/Plack-Request-WithEncoding-0.14.ebuild:
```diff
@@ -0,0 +1,23 @@
+LICENSE="|| ( Artistic GPL-1 GPL-2 GPL-3 )"
+SLOT="0"
+KEYWORDS="~amd64"
+IUSE=""
```
remove

akhuettel commented on dev-perl/Plack-Request-WithEncoding/Plack-Request-WithEncoding-0.14.ebuild:
```diff
@@ -0,0 +1,23 @@
+RDEPEND="dev-perl/HTTP-Message
+	dev-perl/Plack
+	dev-perl/Hash-MultiValue
+	dev-lang/perl"
```
remove

akhuettel commented on dev-perl/Plack-Request-WithEncoding/Plack-Request-WithEncoding-0.14.ebuild:
```diff
@@ -0,0 +1,23 @@
+
+inherit perl-module
+
+DESCRIPTION="Subclass of L<Plack::Request> which supports encoded requests."
```
POD syntax makes no sense here

akhuettel commented on dev-perl/Plack-Request-WithEncoding/Plack-Request-WithEncoding-0.14.ebuild:
```diff
@@ -0,0 +1,23 @@
+	dev-lang/perl"
+
+BDEPEND=">=dev-perl/Module-Build-Tiny-0.39.0
+	dev-perl/Module-Build"
```
does it really need both?

akhuettel commented on dev-perl/Session-Storage-Secure/Session-Storage-Secure-1.000.ebuild:
```diff
@@ -0,0 +1,33 @@
+EAPI=8
+
+DIST_AUTHOR="DAGOLDEN"
+
```
DIST_VERSION=1.000
and the ebuild is 1.0.0

akhuettel commented on dev-perl/Session-Storage-Secure/Session-Storage-Secure-1.000.ebuild:
```diff
@@ -0,0 +1,33 @@
+
+DESCRIPTION="Encrypted, expiring, compressed, serialized session data with integrity"
+
+LICENSE="|| ( Artistic GPL-1 GPL-2 GPL-3 )"
```
remove

akhuettel commented on dev-perl/Session-Storage-Secure/Session-Storage-Secure-1.000.ebuild:
```diff
@@ -0,0 +1,33 @@
+LICENSE="|| ( Artistic GPL-1 GPL-2 GPL-3 )"
+SLOT="0"
+KEYWORDS="~amd64"
+IUSE=""
```
remove

akhuettel commented on dev-perl/Session-Storage-Secure/Session-Storage-Secure-1.000.ebuild:
```diff
@@ -0,0 +1,33 @@
+	dev-perl/Crypt-URandom
+	dev-perl/Sereal-Encoder
+	dev-perl/namespace-clean
+	dev-lang/perl"
```
remove

akhuettel commented on dev-perl/String-Compare-ConstantTime/String-Compare-ConstantTime-0.321.ebuild:
```diff
@@ -0,0 +1,17 @@
+EAPI=8
+
+DIST_AUTHOR="FRACTAL"
+
```
DIST_VERSION=0.321
and the ebuild is 0.321.0

akhuettel commented on dev-perl/String-Compare-ConstantTime/String-Compare-ConstantTime-0.321.ebuild:
remove

akhuettel commented on dev-perl/String-Compare-ConstantTime/String-Compare-ConstantTime-0.321.ebuild:
remove

akhuettel commented on dev-perl/String-Compare-ConstantTime/String-Compare-ConstantTime-0.321.ebuild:
remove

akhuettel commented on dev-perl/String-Random/String-Random-0.32.ebuild:
```diff
@@ -0,0 +1,19 @@
+EAPI=8
+
+DIST_AUTHOR="SHLOMIF"
+
```
DIST_VERSION=0.32
and the ebuild is 0.320.0

akhuettel commented on dev-perl/String-Random/String-Random-0.32.ebuild:
```diff
@@ -0,0 +1,19 @@
+
+DESCRIPTION="Perl module to generate random strings based on a pattern"
+
+LICENSE="|| ( Artistic GPL-1 GPL-2 GPL-3 )"
```
remove

akhuettel commented on dev-perl/String-Random/String-Random-0.32.ebuild:
```diff
@@ -0,0 +1,19 @@
+LICENSE="|| ( Artistic GPL-1 GPL-2 GPL-3 )"
+SLOT="0"
+KEYWORDS="~amd64"
+IUSE=""
```
remove

akhuettel commented on dev-perl/String-Random/String-Random-0.32.ebuild:
```diff
@@ -0,0 +1,19 @@
+KEYWORDS="~amd64"
+IUSE=""
+
+RDEPEND="dev-lang/perl"
```
remove

akhuettel commented on dev-perl/Unicode-Regex-Set/Unicode-Regex-Set-0.04.ebuild:
```diff
@@ -0,0 +1,17 @@
+EAPI=8
+
+DIST_AUTHOR="SADAHIRO"
+
```
DIST_VERSION=0.04
and the ebuild is 0.40.0

akhuettel commented on dev-perl/Unicode-Regex-Set/Unicode-Regex-Set-0.04.ebuild:
```diff
@@ -0,0 +1,17 @@
+
+DESCRIPTION="Subtraction and Intersection of Character Sets in Unicode Regular Expressions"
+
+LICENSE="|| ( Artistic GPL-1 GPL-2 GPL-3 )"
```
remove

akhuettel commented on dev-perl/Unicode-Regex-Set/Unicode-Regex-Set-0.04.ebuild:
```diff
@@ -0,0 +1,17 @@
+LICENSE="|| ( Artistic GPL-1 GPL-2 GPL-3 )"
+SLOT="0"
+KEYWORDS="~amd64"
+IUSE=""
```
remove

akhuettel commented on dev-perl/Unicode-Regex-Set/Unicode-Regex-Set-0.04.ebuild:
```diff
@@ -0,0 +1,17 @@
+KEYWORDS="~amd64"
+IUSE=""
+
+RDEPEND="dev-lang/perl"
```
remove

akhuettel commented on dev-perl/Version-Compare/Version-Compare-0.14.ebuild:
```diff
@@ -0,0 +1,17 @@
+EAPI=8
+
+DIST_AUTHOR="TEX"
+
```
DIST_VERSION=0.14
and the ebuild is 0.140.0

akhuettel commented on dev-perl/Version-Compare/Version-Compare-0.14.ebuild:
```diff
@@ -0,0 +1,17 @@
+
+DESCRIPTION="Compare version strings Version::Compare - Comparing version strings"
+
+LICENSE="|| ( Artistic GPL-1 GPL-2 GPL-3 )"
```
remove

akhuettel commented on dev-perl/Version-Compare/Version-Compare-0.14.ebuild:
```diff
@@ -0,0 +1,17 @@
+LICENSE="|| ( Artistic GPL-1 GPL-2 GPL-3 )"
+SLOT="0"
+KEYWORDS="~amd64"
+IUSE=""
```
remove

akhuettel commented on dev-perl/Version-Compare/Version-Compare-0.14.ebuild:
```diff
@@ -0,0 +1,17 @@
+KEYWORDS="~amd64"
+IUSE=""
+
+RDEPEND="dev-lang/perl"
```
remove

akhuettel reviewed (changes requested):
Main issue, please stick to the Gentoo-specific versions. 
https://wiki.gentoo.org/wiki/Project:Perl/Version-Scheme

thesamesam commented on dev-perl/Workflow/metadata.xml:
```diff
@@ -0,0 +1,23 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE pkgmetadata SYSTEM "https://www.gentoo.org/dtd/metadata.dtd">
+<pkgmetadata>
+<maintainer type="person" proxied="yes">
```
I'd indent these (look at various examples in dev-perl/)

thesamesam commented on dev-perl/Workflow/Workflow-1.61.ebuild:
```diff
@@ -0,0 +1,39 @@
+	>=dev-perl/Log-Log4perl-1.540.0
+	>=dev-perl/Exception-Class-1.450.0
+	>=dev-perl/DateTime-1.540.0
+	dev-lang/perl"
```
You shouldn't need dev-lang/perl as the eclass adds it.

thesamesam commented on dev-perl/Pod-Coverage-TrustPod/Pod-Coverage-TrustPod-0.100005.ebuild:
```diff
@@ -0,0 +1,20 @@
+
+inherit perl-module
+
+DESCRIPTION="allow a module's pod to contain Pod::Coverage hints"
```
Nit:
```suggestion
DESCRIPTION="Allow a module's pod to contain Pod::Coverage hints"
```

thesamesam commented on dev-perl/Mock-MonkeyPatch/Mock-MonkeyPatch-1.02.ebuild:
```diff
@@ -0,0 +1,20 @@
+KEYWORDS="~amd64"
+IUSE=""
+
+DEPEND="dev-lang/perl"
```
```suggestion
```

einhverfr commented on dev-perl/Email-Stuffer/Email-Stuffer-0.018.ebuild:
```diff
@@ -0,0 +1,24 @@
+
+DESCRIPTION="Create and send Email:: emails using a Builder pattern"
+
+LICENSE="|| ( Artistic GPL-1 GPL-2 GPL-3 )"
```
Redundant on the license bit?  Or on the description field?

einhverfr commented:
Ok going through and implementing this feedback.

einhverfr commented:
@akhuettel, when you get a chance, if you could take a look and see if the current modules in the PR have anything else that need to be corrected, I would really appreciate that.  I suspect it is easier for both of us if constructive feedback is given early on in this process.

einhverfr commented:
This is now ready for a review of anything I didn't quite understand yet.

einhverfr commented:
Also if this is too many ebuilds to review at once, I could structure this into (n) smaller PRs.  Just let me know what is easiest.

akhuettel commented on dev-perl/Locales/Locales-0.340.0.ebuild:
```diff
@@ -0,0 +1,20 @@
+
+RDEPEND="dev-perl/String-UnicodeUTF8
+	dev-perl/File-Slurp
+	>=dev-perl/Module-Want-0.6"
```
0.600.0 ??

akhuettel commented on dev-perl/Locales/Locales-0.340.0.ebuild:
```diff
@@ -0,0 +1,20 @@
+	dev-perl/File-Slurp
+	>=dev-perl/Module-Want-0.6"
+
+BDEPEND="test? ( dev-perl/Test-Carp )"
```
tests need runtime deps

akhuettel commented on dev-perl/Log-Any-Adapter-Log4perl/Log-Any-Adapter-Log4perl-0.90.0.ebuild:
```diff
@@ -0,0 +1,17 @@
+KEYWORDS="~amd64"
+
+RDEPEND=">=dev-perl/Log-Log4perl-1.540.0
+	>=dev-perl/Log-Any-1.710.0"
```
tests need runtime deps

akhuettel commented on dev-perl/Mock-MonkeyPatch/Mock-MonkeyPatch-1.20.0.ebuild:
```diff
@@ -0,0 +1,17 @@
+KEYWORDS="~amd64"
+
+BDEPEND=">=dev-perl/Module-Build-Tiny-0.39.0
+	dev-perl/Module-Build"
```
really both?

akhuettel commented on dev-perl/Module-Build-Pluggable-PPPort/Module-Build-Pluggable-PPPort-0.40.0.ebuild:
```diff
@@ -0,0 +1,20 @@
+	dev-perl/Module-Build
+	>=dev-perl/Module-Build-Pluggable-0.10"
+
+BDEPEND="test? ( dev-perl/Test-Requires )"
```
tests need runtime deps

akhuettel commented on dev-perl/Module-Build-Pluggable/Module-Build-Pluggable-0.100.0.ebuild:
```diff
@@ -0,0 +1,21 @@
+	dev-perl/Module-Build
+	dev-perl/Class-Accessor-Lite"
+
+BDEPEND="test? ( dev-perl/Test-SharedFork )"
```
tests need runtime deps

akhuettel commented on dev-perl/Module-Want/Module-Want-0.600.0.ebuild:
```diff
@@ -0,0 +1,19 @@
+KEYWORDS="~amd64"
+
+BDEPEND="dev-perl/Module-Build
+	test? ( dev-perl/Test-Carp )"
```
tests need runtime deps

akhuettel commented on dev-perl/MooX-ClassAttribute/MooX-ClassAttribute-0.011.0.ebuild:
```diff
@@ -0,0 +1,18 @@
+
+RDEPEND="dev-perl/Exporter-Tiny
+	dev-perl/Moo
+	dev-perl/Role-Tiny"
```
dito (not repeating this anymore)

akhuettel commented on dev-perl/OpenOffice-OODoc/OpenOffice-OODoc-2.125.0.ebuild:
```diff
@@ -0,0 +1,17 @@
+KEYWORDS="~amd64"
+
+DEPEND=">=dev-perl/XML-Twig-3.520.0
+	>=dev-perl/Archive-Zip-1.590.0"
```
really DEPEND ?

einhverfr replied:
missed it on my review (tried to do too many in a sitting), will fix and slow down

einhverfr replied:
Out of curiosity. why is DEPEND still supported on recent EAPIs?

akhuettel commented on dev-perl/Plack-Middleware-ReverseProxy/Plack-Middleware-ReverseProxy-0.160.0.ebuild:
```diff
@@ -0,0 +1,17 @@
+
+DESCRIPTION="Supports app to run as a reverse proxy backend"
+
+LICENSE="|| ( Artistic GPL-1 GPL-2 GPL-3 )"
```
looks like you can omit LICENSE setting

akhuettel commented on dev-perl/Workflow/Workflow-1.610.0.ebuild:
```diff
@@ -0,0 +1,37 @@
+	>=dev-perl/XML-Simple-2.250.0
+	dev-perl/Readonly
+	dev-perl/File-Slurp
+	>=dev-perl/Class-Factory-1.06
```
1.60.0 ??

akhuettel reviewed (changes requested):
Looks much better. Some more versions to fix. Apart from that, the most frequent problem is, usually tests in perl modules assume that the runtime dependencies are present!
You have two options there: 
1) either, set `BDEPEND="${RDEPEND} ...."` in general (that's what most of the ebuilds in the tree still do)
2) or add `${RDEPEND}` inside the `test? ()`  clause

einhverfr commented on acct-group/ledgersmb/ledgersmb-0.0.1.ebuild:
```diff
@@ -0,0 +1,9 @@
+# Copyright 2022 Gentoo Authors
```
this is a quick fix.  doing it before I write up my checklist for another pass.

einhverfr commented on app-office/ledgersmb/ledgersmb-1.10.1.ebuild:
```diff
@@ -0,0 +1,153 @@
+dev-perl/Authen-SASL
+dev-perl/CGI-Emulate-PSGI
+dev-perl/Config-IniFiles
+>=dev-perl/Cookie-Baker-0.10
```
yeah I did not always fix the version numbers in the rdepend/bdepend sections it looks like.  I will add this to my checklist fo rthe next pass.

einhverfr commented on app-office/ledgersmb/ledgersmb-1.10.1.ebuild:
```diff
@@ -0,0 +1,153 @@
+	    echo cp -R ${S}/$a ${D}/usr/share/ledgersmb-${VDOTS}
+	    cp -R ${S}/$a ${D}/usr/share/ledgersmb-${VDOTS}  || die "Install Failed"
+	done
+	if ((use systemd))
```
that is probably better.  I get confused with bash boolean evaluation sometimes so went with what I was most sure would work

einhverfr commented:
Ok so my approach here is going to be:

1.  fixup the ledgersmb ebuilds first.
2.  Write up a checklist based on feedback and post it here, and then begin working through all the ebuilds on this.    Given the number of ebuilds I expect this to take at least a week since I want to really make sure on each of them.  If there is any more feedback for this round, please leave it here, I will get incorporate it into the checklist

einhverfr commented on app-office/ledgersmb/ledgersmb-1.9.16.ebuild:
```diff
@@ -0,0 +1,150 @@
+# Copyright 1999-2022 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+# Precalculating version strings for things like initfiles
```
sorry, I moved the logic and left the comment, fixing.

einhverfr commented on app-office/ledgersmb/metadata.xml:
```diff
@@ -0,0 +1,25 @@
+  <longdescription lang="en">The LedgerSMB project provides small and mid-size businesses with open source accounting software: integrating invoicing, order processing, quotations and more (ERP). LedgerSMB aims to provide a strong (multi-currency accounting) basis to build your business on -- without vendor lock-in.</longdescription>
+  <slots>
+    <slot name="1">Version 1.9.x</slot>
+    <slot name="2">Version 1.10.x</slot>
```
I didn't know that was supported, will certainly do that!

einhverfr commented on app-office/ledgersmb/ledgersmb-1.10.1.ebuild:
```diff
@@ -0,0 +1,153 @@
+	VARRAY=($PV)
+	VDOTS="${VARRAY[0]}.${VARRAY[1]}"
+	VUNDER="${VARRAY[0]}_${VARRAY[1]}"h
+	unset IFS;
```
Maybe I should change the SLOT definition to use these and build these variables based on $SLOT?

einhverfr replied:
hmmm probably better as you say to use the ver_cut function here too, and use it in SLOT too.

einhverfr replied:
Please check my latest version here.  I think it is more elegant and much nicer.

einhverfr commented on app-office/ledgersmb/ledgersmb-1.10.1.ebuild:
```diff
@@ -0,0 +1,153 @@
+	done
+	if ((use systemd))
+	   then
+	   insinto /etc/systemd/system/
```
I am not quite sure I understand the two options presented there.  I am going to come back to this after I discuss on #gentoo-systemd.

einhverfr replied:
So reading this and your other comment, it sounds like I need to set up the service file etc and go with the econf option instead.  But I am going to confirm with #gentoo-systemd first.

einhverfr replied:
It sounds like I should just install the systemd.conf file and use econf to point at it unconditionally, right?

einhverfr commented on app-office/ledgersmb/ledgersmb-1.10.1.ebuild:
```diff
@@ -0,0 +1,153 @@
+	)
+xetex? (
+	>=dev-perl/LaTeX-Driver-1.0.0
+	>=dev-perl/Template-Plugin-Latex-3.08
```
I have double checked and normalized all version names in dependencies in the ledgersmb ebuilds. 
I have not done the same to the dependencies yet, but one thing at a time.

einhverfr commented on app-office/ledgersmb/ledgersmb-1.10.1.ebuild:
```diff
@@ -0,0 +1,153 @@
+>=dev-perl/PGObject-Type-ByteString-1.2.3
+dev-perl/PGObject-Util-DBMethod
+>=dev-perl/PGObject-Util-DBAdmin-1.6.1
+>=dev-perl/Plack-1.0031
```
ouch, missed that one, fixing in a bit.

einhverfr commented on app-office/ledgersmb/ledgersmb-1.10.1.ebuild:
```diff
@@ -0,0 +1,153 @@
+	else
+	   exeinto  /etc/init.d
+	   dodir /var/log/ledgersmb
+	   chown ledgersmb:ledgersmb "${D}"/var/log/ledgersmb
```
the current systemd config (from upstream) sends messages to via systemd to the standard system d journal/log. In openrc, they are directed via start-stop-daemon to log files.  This should probably be harmonized at some point.  Leaving open

einhverfr commented on app-office/ledgersmb/ledgersmb-1.10.1.ebuild:
```diff
@@ -0,0 +1,153 @@
+	   newconfd "${S}"/doc/conf/openrc/conf.d/ledgersmb_starman ledgersmb_${VUNDER}
+	fi
+	dodoc -r "${S}"/doc
+}
```
Thanks.  that may in fact help me understand the the systemd docs.

einhverfr commented:
So what I have on my todo list for this next pass is:
1.  Check all BDEPEND and ensure that RDEPEND is included there for testing
2. Double check module versions in RDEPEND and BDEPEND
3. I think I should also edit the commit messages to use gentoo rather than cpan package numbers

In addition I still need to update the ledgersmb build configs to unconditionally set up files etc for both openrc and systemd.

I expect this process to take a while, probably the whole of the next week at least.

einhverfr commented on dev-perl/DBD-Mock/DBD-Mock-1.590.0.ebuild:
```diff
@@ -0,0 +1,20 @@
+RDEPEND=">=dev-perl/Module-Build-Tiny-0.39.0
+	>=dev-perl/DBI-1.643.0"
+
+BDEPEND=">=dev-perl/Test-Exception-0.430.0
```
This is a testing module, so it it is hard to determine what is runtime and what is test.  I assume you are right here but trying to verify.

thesamesam replied:
Take a look at the things it installs and grep them for "Test::Exception" etc?

einhverfr commented:
In this round, I added ${RDEPEND} to BDEPEND following the most common practice in dev-perl.  I also reviewed DEPEND vs RDEPEND, licenses, etc.  And I rewrote commit messages to use gentoo version numbers rather than cpan version numbers.

einhverfr commented:
I just noticed one problem in the ledgersmb metadata.xml in that I updared the slots in the ebuild but missed the metadata.xml.  Will fix this week.

einhverfr commented:
The above problem is corrected, awaiting review

einhverfr commented:
Dependencies checked and updated as needed.  Only starman had changed.

einhverfr commented:
Sorry about spamming the review request.  It is not working the way I expect and the Github UI is acting weird.  I will e filing a bug with Github.com

einhverfr commented:
At this point I am moving the version bumping to separate commits.  If there is a need to change this, I can rebase and  squash commits for the same package together.  But this seems better for collecting feedback and conducting review.

einhverfr commented:
Checking in and wondering what I can do to nudge this along/

thesamesam commented:
Sorry, I'd completely forgot about this. Please ping more aggressively ;)

If still interested, rebase and I can take a look?
